### ISSUE TITLE
How to listen to route change like onChange for React-Router? 

### ISSUE THREAD

**Issue by Dhruvil2104** [Paratron/hookrouter#123]
I know that `react-router` provides `onChange` attribute to listen to route change.

I'm trying to get a log of `users` when they visit a specific route. For e.g. when they visit `/aboutus` route. The user could visit through button click event or entering URL manually or opening a bookmarked page.

There are tracking apps available which tracks counter on button click but I haven't find any app that keeps track of how many times a specific route was visited by the user.

In short, is there a way I can know how many times the user visited a specific route or a way that I can know route change?

**Comment by Paratron** [comment]
Is there any reason to not just install a web based analytics tool like google analytics or the like? It would automatically track visits to a specific route without you having to do anything.

However, if you want to do it by yourself, you need some API to be called whenever the route has been loaded. Lets assume you have such an API in place under `https://example.com/track`. I would create a component somewhere at the root level of the application: 

```
import {usePath} from "hookrouter";

const MyTracker = ({user}) => {
	const path = usePath();

	React.useEffect(() => {
		fetch('https://example.com/track', {
			method: 'POST',
			mode: 'cors',
			body: JSON.stringify({
				path,
				user
			})
		});
	}, [user, path]);

	return null;
};
```
This example component needs to get passed the user object or user id. It uses the `usePath` hook from hookrouter which makes the component re-render whenever the route changes.

The effect will be triggered whenever the route and/or the user changes. It will issue a POST request to the example api and will send the current path and user to the API to be tracked.

**Comment by Dhruvil2104** [comment]
> Is there any reason to not just install a web based analytics tool like google analytics or the like? It would automatically track visits to a specific route without you having to do anything.
> 
> However, if you want to do it by yourself, you need some API to be called whenever the route has been loaded. Lets assume you have such an API in place under `https://example.com/track`. I would create a component somewhere at the root level of the application:
> 
> ```
> import {usePath} from "hookrouter";
> 
> const MyTracker = ({user}) => {
> 	const path = usePath();
> 
> 	React.useEffect(() => {
> 		fetch('https://example.com/track', {
> 			method: 'POST',
> 			mode: 'cors',
> 			body: JSON.stringify({
> 				path,
> 				user
> 			})
> 		});
> 	}, [user, path]);
> 
> 	return null;
> };
> ```
> 
> This example component needs to get passed the user object or user id. It uses the `usePath` hook from hookrouter which makes the component re-render whenever the route changes.
> 
> The effect will be triggered whenever the route and/or the user changes. It will issue a POST request to the example api and will send the current path and user to the API to be tracked.

I'm able to realize what I was doing wrong with the above code of yours. Thanks!

**Comment by Dhruvil2104** [comment]
I'm able to realize what I was doing wrong with the above code yours. Thanks!
